### DISCUSSION TITLE
Config specific country for VPN

### DISCUSSION THREAD

**Discussion by superit63** [engageub/InternetIncome#122] [General]
Hello,
I am switching proxy to using VPNs that I can use for running Mystnode. I couldnt find the way to set up specific location.
P/s: I am using Surfshark VPN

**Comment by engageub** [comment]
Hello,
Do you have OVPN config files?

Thank you

**Reply by superit63** [reply]
I follow this example to set up my VPS account
-e VPN_TYPE=openvpn -e OPENVPN_USER="someusername" -e OPENVPN_PASSWORD="somepassword"

**Reply by engageub** [reply]
Hello,
What about OVPN config file. It should have certificates and other parameters in the file.

Thank you

**Reply by engageub** [reply]
If surfshark, already creates a VPN on the entire network on your Linux Host, then you can use direct connection.
Usually for open vpn, you will have ovpn config file. For wireguard, it may not be required. 

Thank you

**Reply by superit63** [reply]
Sorry for my bad. I am not familar to VPN, the first time I use VPN.
My purpose is running multiple Mystnode, so I will put "-e VPN_TYPE=openvpn -e OPENVPN_USER="someusername" -e OPENVPN_PASSWORD="somepassword" in vpns.txt and multiply it as many lines as possible. Is it right? 
Thanks

**Reply by engageub** [reply]
There are multiple providers for VPN. You can use multiple VPNS by specifying config in each line.
According to the wiki, surfshark does not require ovpn config file.

For surfshark, please use the following wiki. It has SERVER_COUNTRIES option to input country.

Surfshark: https://github.com/qdm12/gluetun-wiki/blob/main/setup/providers/surfshark.md

Other Providers with Config: https://github.com/qdm12/gluetun-wiki/tree/main/setup/providers

Thank you

**Reply by engageub** [reply]
Please use only the following environment variables.
In each line specify the provider name, country name, username and password as mentioned in the following example.

```
-e VPN_SERVICE_PROVIDER=surfshark -e OPENVPN_USER=abc -e OPENVPN_PASSWORD=abc -e SERVER_COUNTRIES=Netherlands
```


Thank you

**Comment by engageub** [comment]
Were you able to successfully run?

Thank you

**Comment by engageub** [comment]
Please update to the latest code if the country name you are using has space in it.
Make sure to  use double quotes if there is a space in the country name.

Thank you

**Reply by superit63** [reply]
Thanks a lot for your help. I will try and get back to you soon. Thanks again

**Reply by engageub** [reply]
Thank you.
Did you get a chance to look into this.
There are multiple VPN providers. 
The script should be able to handle it. 

Thank you

**Reply by superit63** [reply]
Thanks Engageub. It works now, lets see if I got banned or not. I am afraid of VPN :D

**Reply by engageub** [reply]
Hello,
Thank you for the update. Most of the VPN providers use datacenter IP. Very few provide Residential IP.
You will have to select the apps based on the IP type provided by VPN providers. 
Please find the list of VPN providers below which support Datacenter and Residential IP.

https://github.com/engageub/InternetIncome/wiki/Proxies-and-Hosting-Providers


Thank you

**Reply by superit63** [reply]
Hello Engageub,
As I followed the guide, I setup the VPNs like below:
-e VPN_SERVICE_PROVIDER=hidemyass -e OPENVPN_USER='juuma@hotmail.com' -e OPENVPN_PASSWORD='S1I?#6st' -e SERVER_COUNTRIES='USA'
However, I got the issue of AUTH although I have checked the user and pass are correct.
2024-06-05T04:17:07Z WARN HideMyAss dropped support for Linux OpenVPN  so this will likely not work anymore. See https://github.com/qdm12/gluetun/issues/1498.
2024-06-05T04:17:07Z INFO [routing] default route found: interface eth0, gateway 172.17.0.1, assigned IP 172.17.0.2 and family v4
2024-06-05T04:17:07Z INFO [routing] adding route for 0.0.0.0/0
2024-06-05T04:17:07Z INFO [firewall] setting allowed subnets...
2024-06-05T04:17:07Z INFO [routing] default route found: interface eth0, gateway 172.17.0.1, assigned IP 172.17.0.2 and family v4
2024-06-05T04:17:07Z INFO [dns] using plaintext DNS at address 1.1.1.1
2024-06-05T04:17:07Z INFO [http server] http server listening on [::]:8000
2024-06-05T04:17:07Z INFO [firewall] allowing VPN connection...
2024-06-05T04:17:07Z INFO [healthcheck] listening on 127.0.0.1:9999
2024-06-05T04:17:07Z INFO [openvpn] OpenVPN 2.6.8 x86_64-alpine-linux-musl [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD]
2024-06-05T04:17:07Z INFO [openvpn] library versions: OpenSSL 3.1.4 24 Oct 2023, LZO 2.10
2024-06-05T04:17:07Z INFO [openvpn] TCP/UDP: Preserving recently used remote address: [AF_INET]66.115.181.161:553
2024-06-05T04:17:07Z INFO [openvpn] UDPv4 link local: (not bound)
2024-06-05T04:17:07Z INFO [openvpn] UDPv4 link remote: [AF_INET]66.115.181.161:553
2024-06-05T04:17:07Z INFO [openvpn] [server] Peer Connection Initiated with [AF_INET]66.115.181.161:553
2024-06-05T04:17:08Z ERROR [openvpn] AUTH: Received control message: AUTH_FAILED

As I found the discussion on https://github.com/qdm12/gluetun/issues/1498, HMA doesnt support linux anymore.
Please double check and drop HMA in your recommend VPN list.
Thanks

**Comment by engageub** [comment]
Thank you for the update. The warning message mentions that HMA is not supported for Linux as of now.
It has been removed from the list of VPN providers but the setup wiki guide still has the HMA file.

Looking at the code, this is the only provider that has a warning message about dropping support for linux.

Code: https://github.com/qdm12/gluetun/blob/master/internal/configuration/settings/settings.go

Setup Guide: https://github.com/qdm12/gluetun-wiki/tree/main/setup/providers

Updated Providers List: https://github.com/engageub/InternetIncome/wiki/Proxies-and-Hosting-Providers
Updated VPNs File: https://github.com/engageub/InternetIncome/blob/test/vpns.txt

Have you tried using  a simple password for HMA without any special characters. The message shows something related to Authentication. If it does not work, then it is likely not supported anymore. 



Thank you

**Comment by superit63** [comment]
Hi Engageub,

I have tried simple password also, and as I checked on official website of
HMA, they dont support Linux.


Vào Th 4, 5 thg 6, 2024 vào lúc 13:07 engageub ***@***.***>
đã viết:

> Thank you for the update. The warning message mentions that HMA is not
> supported for Linux as of now.
> It has been removed from the list of VPN providers but the setup wiki
> guide still has the HMA file.
>
> Looking at the code, this is the only provider that has a warning message
> about dropping support for linux.
>
> Code:
> https://github.com/qdm12/gluetun/blob/master/internal/configuration/settings/settings.go
>
> Setup Guide:
> https://github.com/qdm12/gluetun-wiki/tree/main/setup/providers
>
> Updated Providers List:
> https://github.com/engageub/InternetIncome/wiki/Proxies-and-Hosting-Providers
> Updated VPNs File:
> https://github.com/engageub/InternetIncome/blob/test/vpns.txt
>
> Have you tried using a simple password for HMA without any special
> characters. The message shows something related to Authentication. If it
> does not work, then it is likely not supported anymore.
>
> Thank you
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/engageub/InternetIncome/discussions/122#discussioncomment-9673589>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BIN6POPP3NSVHGUNNMAMA7LZF2TLRAVCNFSM6AAAAABIMQKGZSVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TMNZTGU4DS>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Tran Quang Sieu

**Reply by engageub** [reply]
@superit63 

Thank you for the confirmation.

**Comment by engageub** [comment]
Hello,
There is a bug in the latest version of gluetun which is causing high cpu utilization. Please download the latest version of InternetIncome Script from test branch if you would like to use VPN.
Older Stable version has been added.
Thank you
